Rsyslog coredump in omfile  version 3.39.0

It's probably the issue where if dynafile creation fails.

log:
7975.233516081:testWriteData:Reg/w0: rainerscript.c:     ACTION 0 [mmnormalize:action(type="mmnormalize" ...)]
7975.233553196:testWriteData:Reg/w0: ruleset.c: executing action 0
7975.233574287:testWriteData:Reg/w0: ../action.c: action 'action-0-mmnormalize': called, logging to mmnormalize (susp 0/0, direct q 1)
7975.233603540:testWriteData:Reg/w0: ../action.c: wti 0x1ae9ca0: we need to create a new action worker instance for action 0
7975.233624825:testWriteData:Reg/w0: ../action.c: wti 0x1ae9ca0: created action worker instance 1 for action 0
7975.233649646:testWriteData:Reg/w0: ../action.c: action[action-0-mmnormalize] transitioned to state: itx
7975.233664874:testWriteData:Reg/w0: ../action.c: entering actionCalldoAction(), state: itx, actionNbr 0
7975.233699821:testWriteData:Reg/w0: ../action.c: action[action-0-mmnormalize] transitioned to state: rdy
7975.233719601:testWriteData:Reg/w0: ../action.c: actionCommit[action-0-mmnormalize]: enter, 0 msgs
7975.233736982:testWriteData:Reg/w0: ../action.c: actionCommit[action-0-mmnormalize]: done, iRet 0
7975.233751921:testWriteData:Reg/w0: ../action.c: action 'action-0-mmnormalize': set suspended state to 0
7975.233771278:testWriteData:Reg/w0: rainerscript.c:     IF
7975.233816407:testWriteData:Reg/w0: rainerscript.c:         var 'parsesuccess'
7975.233878725:testWriteData:Reg/w0: rainerscript.c:       ==
7975.233929376:testWriteData:Reg/w0: rainerscript.c:         string 'OK'
7975.234007145:testWriteData:Reg/w0: rainerscript.c: eval expr 0x1af2cc0, type 'CMP_EQ'
7975.234040198:testWriteData:Reg/w0: rainerscript.c: eval expr 0x1af2bd0, type 'V'
7975.234065281:testWriteData:Reg/w0: rainerscript.c: rainerscript: (string) var 23: 'OK'
7975.234085475:testWriteData:Reg/w0: rainerscript.c: eval expr 0x1af2bd0, return datatype 'S':0
7975.234103845:testWriteData:Reg/w0: rainerscript.c: eval expr 0x1af2cc0, return datatype 'N':1
7975.234121386:testWriteData:Reg/w0: ruleset.c: if condition result is 1
7975.234135482:testWriteData:Reg/w0: rainerscript.c:     ACTION 1 [builtin:omfile:action(type="builtin:omfile" ...)]
7975.234170775:testWriteData:Reg/w0: ruleset.c: executing action 1
7975.234189441:testWriteData:Reg/w0: ../action.c: action 'action-1-builtin:omfile': called, logging to builtin:omfile (susp 0/0, direct q 1)
Segmentation fault

when I INSTALL VAlgring, there are log:

checking for a supported OS... ok (linux)
checking for the kernel version... unsupported (3.X.X_X-X-X-X)
configure: error: Valgrind works on kernels 2.4, 2.6

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -148,6 +148,7 @@ TESTS_DEFAULT = \
 	omusrmsg_ratelimit_name.sh \
 	omfile-module-params.sh \
 	omfile-dynafilecachesize-invalid.sh \
+	omfile-dynafile-mmnormalize-property.sh \
 	omfile-read-only-errmsg.sh \
 	omfile-null-filename.sh \
 	omfile-whitespace-filename.sh \

--- a/tests/omfile-dynafile-mmnormalize-property.sh
+++ b/tests/omfile-dynafile-mmnormalize-property.sh
@@ -1,0 +1,44 @@
+#!/bin/bash
+# Regression reproducer for issue #3488: mmnormalize populates a JSON property
+# used by an omfile dynafile template. The old crash happened while resolving
+# that property during action processing. The oracle is clean shutdown and the
+# expected dynafile write; any SIGSEGV or failed template lookup fails the test.
+. ${srcdir:=.}/diag.sh init
+require_plugin imtcp
+require_plugin mmnormalize
+
+generate_conf
+add_conf '
+module(load="../plugins/imtcp/.libs/imtcp")
+module(load="../plugins/mmnormalize/.libs/mmnormalize")
+
+input(type="imtcp" port="0" listenPortFileName="'$RSYSLOG_DYNNAME'.tcpflood_port" ruleset="writeData")
+template(name="LogFormat" type="string" string="%msg%\n")
+template(name="WriteFile" type="string" string="'$RSYSLOG_DYNNAME'.%$!clientIP%.log")
+
+ruleset(name="writeData"
+        queue.type="fixedArray"
+        queue.size="10000"
+        queue.dequeueBatchSize="2048"
+        queue.workerThreads="2"
+        queue.workerThreadMinimumMessages="10000") {
+    action(type="mmnormalize" rule="rule=:%clientIP:ipv4% %other:rest%" userawmsg="on")
+    if $parsesuccess == "OK" then {
+        action(type="omfile" DynaFile="WriteFile" template="LogFormat")
+    }
+}
+'
+
+startup
+tcpflood -m1 -O -M '"192.0.2.15 - - \"GET / HTTP/1.1\" 200 17 \"-\" \"-\" \"-\""'
+shutdown_when_empty
+wait_shutdown
+
+printf ' - "GET / HTTP/1.1" 200 17 "-" "-" "-"\n' > "${RSYSLOG_DYNNAME}.expected"
+cmp_exact_file "${RSYSLOG_DYNNAME}.expected" "${RSYSLOG_DYNNAME}.192.0.2.15.log" || {
+    echo "FAIL: dynafile write did not use the normalized clientIP property"
+    cat "${RSYSLOG_DYNNAME}.192.0.2.15.log"
+    error_exit 1
+}
+
+exit_test

--- a/tests/omfile-dynafile-mmnormalize-property.sh
+++ b/tests/omfile-dynafile-mmnormalize-property.sh
@@ -22,7 +22,7 @@ ruleset(name="writeData"
         queue.dequeueBatchSize="2048"
         queue.workerThreads="2"
         queue.workerThreadMinimumMessages="10000") {
-    action(type="mmnormalize" rule="rule=:%clientIP:ipv4% %other:rest%" userawmsg="on")
+    action(type="mmnormalize" rule="rule=:<%pri:number%>%clientIP:ipv4% %other:rest%" userawmsg="on")
     if $parsesuccess == "OK" then {
         action(type="omfile" DynaFile="WriteFile" template="LogFormat")
     }
@@ -30,7 +30,7 @@ ruleset(name="writeData"
 '
 
 startup
-tcpflood -m1 -O -M '"192.0.2.15 - - \"GET / HTTP/1.1\" 200 17 \"-\" \"-\" \"-\""'
+tcpflood -m1 -O -M "'<13>192.0.2.15 - - \"GET / HTTP/1.1\" 200 17 \"-\" \"-\" \"-\"'"
 shutdown_when_empty
 wait_shutdown
 

--- a/tests/omfile-dynafile-mmnormalize-property.sh
+++ b/tests/omfile-dynafile-mmnormalize-property.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Regression reproducer for issue #3488: mmnormalize populates a JSON property
 # used by an omfile dynafile template. The old crash happened while resolving
 # that property during action processing. The oracle is clean shutdown and the
@@ -30,7 +30,7 @@ ruleset(name="writeData"
 '
 
 startup
-tcpflood -m1 -O -M "'<13>192.0.2.15 - - \"GET / HTTP/1.1\" 200 17 \"-\" \"-\" \"-\"'"
+tcpflood -m1 -O -M "\"<13>192.0.2.15 - - \\\"GET / HTTP/1.1\\\" 200 17 \\\"-\\\" \\\"-\\\" \\\"-\\\"\""
 shutdown_when_empty
 wait_shutdown
 

